deployments/pathmap-migrator/src/main/java/org/commonjava/indy/pathmap/migrate/ScanCmd.java

```diff
@@ -104,41 +110,32 @@ public void run( MigrateOptions options )
     }
 
     private List<String> listValidPkgFolders( final String baseDir )
-            throws IOException
     {
-        final Predicate<Path> validPkgFolderPredicate =
-                p -> Files.isDirectory( p ) && PackageTypeConstants.isValidPackageType(
-                        p.getName( p.getNameCount() - 1 ).toString() );
-        return listPaths( baseDir, 1, 3, validPkgFolderPredicate );
+        final List<String> pkgPaths = new ArrayList<>( 3 );
+        for ( String pkg : Arrays.asList( PKG_TYPE_GENERIC_HTTP, PKG_TYPE_MAVEN, PKG_TYPE_NPM ) )
+        {
+            if ( Files.isDirectory( Paths.get( baseDir, pkg ) ) )
+            {
+                pkgPaths.add( Paths.get( baseDir, pkg ).toString() );
+            }
+        }
+        return pkgPaths;
     }
 
-    //    private List<String> listReposPath( final String pkgDir )
-    //            throws IOException
-    //    {
-    //        final List<String> allReposPath =
-    //                listPaths( pkgDir, 1, 2000, p -> Files.isDirectory( p ) && !p.equals( Paths.get( pkgDir ) ) );
-    //        System.out.println(
-    //                String.format( "There are %s repos in the pkg folder %s to migrate", allReposPath.size(), pkgDir ) );
-    //        return allReposPath;
-    //    }
-
     private int listFiles( final String pkgDir, final MigrateOptions options )
             throws IOException
     {
         final Path repoPath = Paths.get( pkgDir );
         final String pkgName = repoPath.getName( repoPath.getNameCount() - 1 ).toString();
-        //        final String repoName = repoPath.getName( repoPath.getNameCount() - 1 ).toString();
         final String todoPrefix = TODO_FILES_DIR + "-" + pkgName;
         System.out.println( String.format( "Start to scan package %s for files", pkgDir ) );
         final List<String> filePaths = new ArrayList<>( options.getBatchSize() );
         final AtomicInteger batchNum = new AtomicInteger( 0 );
         final AtomicInteger totalFileNum = new AtomicInteger( 0 );
-        //        final List<String> generatedToDoFiles = Collections.synchronizedList( new ArrayList<>( 200 ) );
         Files.walk( repoPath, Integer.MAX_VALUE ).filter( Files::isRegularFile ).forEach( p -> {
             filePaths.add( p.toString() );
```

Comment:
Can we add a filter to exclude all "http-metadata.json" files? If they are just caches, we may want to ignore them.

Reply:
Sounds good. I'll add one more option to let us decide if filter http-metadata.